hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/SCMCommonPlacementPolicy.java

```diff
@@ -503,4 +508,72 @@ public Set<ContainerReplica> replicasToCopyToFixMisreplication(
   protected Node getPlacementGroup(DatanodeDetails dn) {
     return nodeManager.getClusterNetworkTopologyMap().getAncestor(dn, 1);
   }
+
+  @Override
+  public Set<ContainerReplica> replicasToRemoveToFixOverreplication(
```

Comment:
There are no comments or description about what algorithm is being used to remove the replicas and how it also ensure the placement policy is not violated etc. Can you please describe the algorithm used here, perhaps in a javadoc comment above the method and add some comments indicating what is going on in the code.

Reply:
Have added the logic in javadoc